client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/plan/index.tsx

```diff
@@ -52,13 +55,15 @@ const getPlanTaskSubtitle = (
 };
 
 const getPlanSelectedTask: TaskAction = ( task, flow, context ): Task => {
-	const {
-		siteInfoQueryArgs,
-		displayGlobalStylesWarning,
-		shouldDisplayWarning,
-		globalStylesMinimumPlan,
-		isVideoPressFlowWithUnsupportedPlan,
-	} = context;
+	const { siteSlug, displayGlobalStylesWarning, globalStylesMinimumPlan, planCartItem, site } =
+		context;
+
+	const productSlug = planCartItem?.product_slug ?? site?.plan?.product_slug;
+
+	const isVideoPressFlowWithUnsupportedPlan =
+		isVideoPressFlow( flow ) && ! planHasFeature( productSlug as string, FEATURE_VIDEO_UPLOADS );
+
+	const shouldDisplayWarning = displayGlobalStylesWarning || isVideoPressFlowWithUnsupportedPlan;
```

Comment:
These variables are only used inside this task, so I moved from task-helper to here.